test_problems/pureFluidTest/testPureWater.cpp

```diff
@@ -24,32 +24,29 @@ int main()
     double pres;
     try {
 
-        XML_Node* xc = get_XML_File("liquidvapor.xml");
-        XML_Node* const xs = xc->findNameID("phase", "water");
-        ThermoPhase* water_tp = newPhase(*xs);
-        PureFluidPhase* w = dynamic_cast <PureFluidPhase*>(water_tp);
+        unique_ptr<ThermoPhase> w(newPhase("liquidvapor.yaml", "water"));
```

Comment:
@speth For my own education, why is this better than using `PureFluidPhase w`?

Reply:
It's more similar to the method of instantiating the phase that was being used before, in that it allocates the object on the heap and you just have a pointer to it. I prefer it in this case because (a) it means you don't have to change all the pointer dereferencing operations (`w->foo`) to member access (`w.foo`), and (b) there is no point in this test where the fact that this  is a `PureFluidPhase` object is actually used -- all of the member functions called are ones that are defined in the base class, so it's fine if all we have is a pointer to a `ThermoPhase`.

Reply:
Awesome, thanks for the explanation!